geps/gep-2694.md

```diff
@@ -0,0 +1,108 @@
+# GEP-2694: Support
+
+* Issue: [#2694](https://github.com/kubernetes-sigs/gateway-api/issues/2694)
+* Status: Provisional
+
+## TLDR
+
+This GEP proposes adding regex support for path **redirects and rewrites**.
+
+## Goals
+
+* Replace with a Regular Expression substitution
+
+## API
+
+Although not all gateway APIs support the same regex, it is important
+to find a more commonly supported regex and establish a specification.
+
+* Path redirects
+* Host rewrites
```

Comment:
If we have conformance tests, that moves the feature to Extended (that's the definition difference between ImplementationSpecific and Extended).

Practically, it may not be that big a deal to have an Extended feature that says "supports <some flavor> of RegularExpression", as then implementations can just not implement it if they want.

Reply:
What I was suggesting is that we _could_ have some "incomplete" tests if we wanted, but I do not believe we should be aiming for extended, I feel very strongly we should be targeting implementation-specific. If you're saying you wont accept tests for something implementation specific because it would fly in the face of the usual, then I suppose we can avoid even lightweight tests. I'm flexible on the details, except that I feel strongly extended should be saved for a later iteration, if ever.

Reply:
Right now, we _define_ Extended as meaning "optional, but with a defined behavior that is enforced with conformance tests". If we have an ImplementationSpecific feature with conformance tests, there's no way for people to know that the feature is Implementation Specific, since this situation hasn't arisen before. I guess I don't understand what you're aiming for Implementation Specific to mean here. This seems like a great example of where conformance tests work well - because we can say "this feature means supporting `RE2` (or whatever) flavor regexes, but it's optional".

It's not that we _can't_ do it, it's just that it doesn't make sense under the feature support levels we support today.

If your concern is about Extended implying more stability than is actually there, there's no reason why an Extended feature has to stay - we can have an Extended feature in Experimental that gets removed. Again, we just haven't done that before.

I'm reluctant to change what Implementation Specific means without understanding what you're trying to achieve more though.

Reply:
That's fine.